JetNews/app/src/main/java/com/example/jetnews/ui/interests/InterestsScreen.kt

```diff
@@ -442,13 +466,67 @@ private fun TopicItem(itemTitle: String, selected: Boolean, onToggle: () -> Unit
  * Full-width divider for topics
  */
 @Composable
-private fun TopicDivider() {
+private fun TopicDivider(modifier: Modifier = Modifier) {
     Divider(
-        modifier = Modifier.padding(start = 90.dp, top = 8.dp, bottom = 8.dp),
+        modifier = modifier.padding(start = 90.dp, top = 8.dp, bottom = 8.dp),
         color = MaterialTheme.colors.onSurface.copy(alpha = 0.1f)
     )
 }
 
+/**
+ * TabRow for the InterestsScreen
+ */
+@Composable
+private fun InterestsTabRow(
+    selectedTabIndex: Int,
+    updateSection: (Sections) -> Unit,
+    tabContent: List<TabContent>
+) {
+    // TODO: Make the tabs narrower and aligned to the left in large screens
+    TabRow(
+        selectedTabIndex = selectedTabIndex,
+        backgroundColor = MaterialTheme.colors.onPrimary,
+        contentColor = MaterialTheme.colors.primary
+    ) {
+        tabContent.forEachIndexed { index, tabContent ->
+            val colorText = if (selectedTabIndex == index) {
+                MaterialTheme.colors.primary
+            } else {
+                MaterialTheme.colors.onSurface.copy(alpha = 0.8f)
+            }
+            Tab(
+                selected = selectedTabIndex == index,
+                onClick = { updateSection(tabContent.section) },
+                modifier = Modifier.heightIn(min = 48.dp)
+            ) {
+                Text(
+                    text = stringResource(id = tabContent.section.titleResId),
+                    color = colorText,
+                    style = MaterialTheme.typography.subtitle2,
+                    modifier = Modifier.paddingFromBaseline(top = 20.dp)
+                )
+            }
+        }
+    }
+}
+
+/**
+ * Returns the max width for a Topic Item given maxWidth and number of columns constraints.
+ */
+@Composable
+private fun rememberItemMaxWidth(windowMaxWidth: Dp, columns: Int) =
+    remember(windowMaxWidth, columns) {
```

Comment:
Are these `remember` a bit of a premature optimization? It seems like this logic should be pretty simple to compute, and it avoids forgetting a necessary key here.

Reply:
I don't think they are to be honest. Sure, the logic is simple to compute but even like that, I feel like that logic shouldn't be re-executed on every recomposition (as that could happen quite often). As the result of this computation is not likely to change often, I prefer to `remember` it